minio/api.py

```diff
@@ -839,29 +874,49 @@ def _do_put_object(self, bucket_name, object_name, data,
 
     def _stream_put_object(self, bucket_name, object_name, data,
                            data_content_size, data_content_type='application/octet-stream'):
+        """
+        Streaming multipart upload operation.
+
+        :param bucket_name: Bucket name of the multipart upload.
+        :param object_name: Object name of the multipart upload.
+        :param data_content_size: Total size of the content to be uploaded.
+        :param data_content_type: Content type of of the multipart upload.
+           Defaults to 'application/octet-stream'.
+        """
         part_size = calculate_part_size(data_content_size)
         current_uploads = ListIncompleteUploadsIterator(self._http,
                                                         self._endpoint_url,
                                                         bucket_name,
                                                         object_name,
                                                         access_key=self._access_key,
-                                                        secret_key=self._secret_key)
+                                                        secret_key=self._secret_key,
+                                                        region=self._get_region(bucket_name))
         upload_id = None
         for upload in current_uploads:
             if object_name == upload.object_name:
                 upload_id = upload.upload_id
 
-        ## Initialize variables
+        # Initialize variables
         uploaded_parts = {}
         uploaded_etags = []
         total_uploaded = 0
         current_part_number = 1
 
-        ## If upload_id is None its a new multipart upload.
+        # Some streams wouldn't implement seek() method.
+        # Handle it and set it to False.
+        try:
+            data.seek(0)
+        except:
```

Comment:
> its not a good idea to ignore any exception.

It was on purpose to know if a particular method exists, if it doesn't then we fall back to a different method to upload. I will find a better way to know this.